### PR TITLE
Fixed scrolling 

### LINKED ISSUE #9: Scrolling wheelScale
Scrolling does not seem to work here? But I think it should since its a part of the code (havent looked in deeply)

### PR DESCRIPTION
**Pull request by z3t0**
Fixed #9

### CODE REVIEW

**Comment by z3t0** [comment]
Uses inventoryWidth

**Comment by z3t0** [comment]
accidentally close

**Comment by z3t0** [comment]
@deathcap could you merge this so we can continue updating the package.json in voxelmetaverse

**Comment by z3t0** [comment]
Only scrolls when chat bar is closed

File: hotbar.coffee [hunk 1]
```diff
@@ -47,6 +47,8 @@ class InventoryHotbarClient extends InventoryHotbarCommon
     @keys = game.plugins.get('voxel-keys') ? throw new Error('voxel-inventory-hotbar requires voxel-keys plugin')
     @wheelEnable = opts.wheelEnable ? false # enable scroll wheel to change slots?
     @wheelScale = opts.wheelScale ? 1.0  # mouse wheel scrolling sensitivity
+    @console = game.plugins.get('voxel-console')
+    @scrollLock = true; # allows scrolling on true
```
Comment (deathcap) [inline hotbar.coffee]: Unnecessary semicolon; not required and should not normally be used in CoffeeScript

File: hotbar.coffee [hunk 2]
```diff
@@ -92,16 +94,28 @@ class InventoryHotbarClient extends InventoryHotbarCommon
   enable: () ->
     @inventoryWindow.container.style.visibility = ''
     @onSlots = {}
+    @console.on 'open', =>
```
Comment (deathcap) [inline hotbar.coffee]: Note to self: requires https://github.com/deathcap/voxel-console/pull/3. I'll review that PR first

File: hotbar.coffee (hunk 1 above)
Comment (z3t0) [inline hotbar.coffee]: My bad.

**Comment by z3t0** [comment]
Removed ;

File: hotbar.coffee [hunk 3]
```diff
@@ -92,16 +94,28 @@ class InventoryHotbarClient extends InventoryHotbarCommon
...
+      @scrollLock = false;
```
Comment (deathcap) [inline hotbar.coffee]: Actually, I think there's a better way to do this. Try replacing your `@scrollLock` and voxel-console events with `@game.shell.pointerLock` - it seems to already do what you want here, and is more general than just having the console open. For example, it also takes into account when another modal dialog is open (e.g., inventory) - it is only true when the browser has acquired pointer lock (to interact with the game with mouse look).

File: hotbar.coffee [hunk 4]
```diff
@@ -92,16 +94,28 @@ class InventoryHotbarClient extends InventoryHotbarCommon
...
-        delta = Math.floor delta
+        delta = Math.floor delta // 120
+        if delta > 0
+          delta = 1
+        else if delta < 0
+          delta = -1
         newSlot = @selectedIndex + delta
-        newSlot = newSlot %% @inventoryWindow.width
-        console.log newSlot
+        if(newSlot == @inventoryWindow.width)
```
Comment (deathcap) [inline hotbar.coffee]: Was `newSlot = newSlot %% @inventoryWindow.width` not sufficient here? It's supposed to wrap around like your `if` statements

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "voxel-inventory-hotbar",
   "description": "a hotbar for player inventory (voxel.js plugin)",
-  "version": "0.5.0",
```
Comment (deathcap) [inline package.json]: Remove this version change, I'll bump it after merge

File: hotbar.coffee (hunk 2 above)
Comment (z3t0) [inline hotbar.coffee]: For this we could have that be a soft dependency

File: hotbar.coffee (hunk 4 above)
Comment (z3t0) [inline hotbar.coffee]: Oh wait i see, I didnt understand what that was doing before. I'll do some testing

**Comment by deathcap** [comment]
Tested this change (with a few modifications: removing voxel-console dependency, setting wheelEnable) but with the mouse I'm using (Apple Magic Mouse), scrolling the mouse still uncontrollably flips through the slots. This may be an inherent problem with this mouse, I know Minecraft has the same problem with it: https://bugs.mojang.com/browse/MC-42367 - so all I've been able to do is disable scrolling, but it does work with a Microsoft IntelliSense mouse (with a discrete scroll wheel instead of the touch-sensitive interface).

**Comment by z3t0** [comment]
It worked for me with a normal mouse as well. I'll see if I can find some
way to make it work with the Magic Mouse.

On Tue, Apr 14, 2015 at 8:59 PM, deathcap <notifications@github.com> wrote:

> Tested this change (with a few modifications: removing voxel-console
> dependency, setting wheelEnable) but with the mouse I'm using (Apple Magic
> Mouse), scrolling the mouse still uncontrollably flips through the slots.
> This may be an inherent problem with this mouse, I know Minecraft has the
> same problem with it: https://bugs.mojang.com/browse/MC-42367 - so all
> I've been able to do is disable scrolling, but it does work with a
> Microsoft IntelliSense mouse (with a discrete scroll wheel instead of the
> touch-sensitive interface).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deathcap/voxel-inventory-hotbar/pull/10#issuecomment-93166919>
> .
>



-- 
Thanks,
Rafi Khan

**Comment by z3t0** [comment]
@deathcap Are you talking about #11 working for you?

**Pull request closed without merging**